app/models/ambassador.rb

```diff
@@ -0,0 +1,41 @@
+class Ambassador < SimpleDelegator
+  def self.all
+    decorate(User.ambassadors)
+  end
+
+  def self.decorate(target)
+    unless target.is_a?(User) || target.respond_to?(:all?) && target.all? { |e| e.is_a?(User) }
+      raise ArgumentError, "Must be a User or collection of Users"
+    end
+
+    Array(target).map { |u| new(u) }
+  end
+
+  def initialize(user)
+    unless user.is_a?(User) && user.ambassador?
+      raise ArgumentError, "Must be an ambassador"
+    end
+    super(user)
+  end
+
+  def percent_complete
+    return 0.0 if ambassador_task_assignments.empty?
+    (completed_tasks_count / tasks_count.to_f).round(2)
+  end
+
+  def progress_count
+    "#{completed_tasks_count}/#{tasks_count}"
+  end
+
+  def completed_tasks_count
+    ambassador_task_assignments.select(&:completed?).count
+  end
+
+  def tasks_count
+    ambassador_task_assignments.count
+  end
+
+  def groups
+    Organization.where(id: memberships.select(:organization_id))
```

Comment:
`select` isn't like sequel here, it will instantiate all the memberships records. I think it would be better with `membership.pluck("DISTINCT organization_id")`

Reply:
Actually, https://github.com/bikeindex/bike_index/pull/760/files#r284909158 is a better solution